plugins/ipam/host-local/config.go

```diff
@@ -39,31 +39,29 @@ type IPAMArgs struct {
 }
 
 type Net struct {
-	Name string      `json:"name"`
-	IPAM *IPAMConfig `json:"ipam"`
+	Name  string      `json:"name"`
+	IPAM  *IPAMConfig `json:"ipam"`
+	IPAM6 *IPAMConfig `json:"ipam6"`
 }
 
 // NewIPAMConfig creates a NetworkConfig from the given network name.
-func LoadIPAMConfig(bytes []byte, args string) (*IPAMConfig, error) {
+func LoadIPAMConfig(bytes []byte, args string) (*IPAMConfig, *IPAMConfig, error) {
 	n := Net{}
 	if err := json.Unmarshal(bytes, &n); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	if args != "" {
 		ipamArgs := IPAMArgs{}
 		err := types.LoadArgs(args, &ipamArgs)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 	}
 
-	if n.IPAM == nil {
-		return nil, fmt.Errorf("%q missing 'ipam' key")
+	if n.IPAM == nil && n.IPAM6 == nil {
+		return nil, nil, fmt.Errorf("Need at least one of ipam or ipam6 key")
 	}
 
-	// Copy net name into IPAM so not to drag Net struct around
```

Comment:
What @dcbw is saying that we have no type checks at all to distinguish and check if the given IP information is appropriate for the respective fields. Before this PR, it is possible to pass both IPv4 and IPv6 information within IPAM and `host-local` will happily allocate an address. And after this PR it will be possible to pass both versions to both fields, which should not be the case anymore as we now want to distinguish input and output.